### PR TITLE
LibCrypto: Add Chacha20Poly1305

### PR DESCRIPTION
**Pull request by stelar7**

### CODE REVIEW

File: Userland/Libraries/LibCrypto/AEAD/ChaCha20Poly1305.cpp [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        mac_data.append(0);
+
+    u8 placeholder[8] = { 0 };
+    // The length of the additional data in octets (as a 64-bit little-endian integer).
+    mac_data.append(ReadonlyBytes { placeholder, 8 });
+    ByteReader::store(static_cast<u8*>(mac_data.end_pointer()) - sizeof(u64), AK::convert_between_host_and_little_endian(static_cast<u64>(aad.size())));
+
+    // The length of the ciphertext in octets (as a 64-bit little-endian integer).
+    mac_data.append(ReadonlyBytes { placeholder, 8 });
+    ByteReader::store(static_cast<u8*>(mac_data.end_pointer()) - sizeof(u64), AK::convert_between_host_and_little_endian(static_cast<u64>(ciphertext.size())));
```
Comment (alimpfard) [inline Userland/Libraries/LibCrypto/AEAD/ChaCha20Poly1305.cpp:73]: `mac_data.end_pointer()` should be a `u8*` already, is it not?

**Review by alimpfard — COMMENTED** [review]
basically LGTM.

File: Userland/Libraries/LibCrypto/AEAD/ChaCha20Poly1305.cpp (hunk 1 above)
Comment (stelar7) [reply Userland/Libraries/LibCrypto/AEAD/ChaCha20Poly1305.cpp:73]: Its a `void*` 🤷

Comment (alimpfard) [reply Userland/Libraries/LibCrypto/AEAD/ChaCha20Poly1305.cpp:73]: huh. well okay.

**Pull request merged**